hs_core/tasks.py

```diff
@@ -387,14 +387,20 @@ def add_zip_file_contents_to_resource(pk, zip_file_path):
         files = zcontents.get_files()
 
         resource.file_unpack_status = 'Running'
-        resource.save()
+        resource.save(update_fields=['file_unpack_status'])
 
+        resource_files = []
         for i, f in enumerate(files):
             logger.debug("Adding file {0} to resource {1}".format(f.name, pk))
-            utils.add_file_to_resource(resource, f)
+            res_file = utils.add_file_to_resource(resource, f, save_file_system_metadata=False)
+            res_file.set_system_metadata(resource=resource, save=False)
+            resource_files.append(res_file)
             resource.file_unpack_message = "Imported {0} of about {1} file(s) ...".format(
                 i, num_files)
-            resource.save()
+            resource.save(update_fields=['file_unpack_message'])
+
+        ResourceFile.objects.bulk_update(resource_files, ResourceFile.system_meta_fields(),
+                                         batch_size=settings.BULK_UPDATE_CREATE_BATCH_SIZE)
```

Comment:
could combine the two `resource.save()` statements (L390 and L400) into a single query by calling
`resource.save(update_fields=['file_unpack_status', 'file_unpack_message'])`
after the `for i, f in enumerate(files):` loop

Reply:
By combining we will save only one db query. Then in `resource.save()` (L400) we have to update the field `file_unpack_status` per file even though the value for this field is not changing per file.

Reply:
sounds good